app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx

```diff
@@ -127,8 +128,8 @@ export default function ConnectionsTabs({
                 }
                 value={connectorType.name}
                 disableTouchRipple
-                key={`${connectorType.name}=${connectorTypeIndex}`}
-                id={connectorType.name}
+                key={`${connectorType.name}-${connectorTypeIndex}`}
+                id={`connectionlist-connectiontabs-tabs-loop-${index}-${connectorTypeIndex}`}
```

Comment:
you don't need to specify indices in the test tags (if `id` is solely used a test tag, in that case rename to `data-testid`). There are ways to locate n-th element of a list for example. Please see the examples here:
https://stackoverflow.com/questions/33939198/selenium-webdriver-finding-elements-using-cssselector-and-nth-child

Reply:
changed id to data-testid